sources/common/src/ISignal.cpp

```diff
@@ -32,6 +32,27 @@
 
 #include "ISignal.h"
 
+SIGNAL::ISignal::ISignal(const std::string &name,
+                         size_t size,
+                         const TIME::Timestamp &rTimestamp,
+                         uint64_t value,
+                         SourceRegistry::HandleT sourceHandle) :
+    Signal(name, size, rTimestamp, "wire", sourceHandle),
+    m_Value(value)
+{
+    if (size > 64)
+    {
+        throw EXCEPTION::VcdException(EXCEPTION::Error::VECTOR_SIZE_EXCEEDED,
+                                      "Vectors sizes greater than 64 bits are not allowed. Requested " +
+                                      std::to_string(size) + "-bit size.");
+    }
+
+    if ((value > (std::numeric_limits<uint64_t>::max() >> (64 - size))) || (0 == size))
```

Comment:
It can still cause undefined behavior. `0 == size` must be done first in the `or`.

Reply:
Just changed that. Does it mean that `std::numeric_limits<uint64_t>::max() >> 64` generates undefined behavior? It is not 0?

Reply:
No, it's undefined.

8.8/1:
> (...)The behavior is undefined if the right operand is negative, or greater than or equal to the length in bits of the promoted left operand.

64 is equal to `uint64_t` length in bits hence it's undefined.

Reply:
Good to know. Thanks!